pylon_camera/include/pylon_camera/internal/impl/pylon_camera_gige.hpp

```diff
@@ -239,35 +236,135 @@ bool PylonGigECamera::applyCamSpecificStartupSettings(const PylonCameraParameter
                 // package size * n_cams + 5% overhead = inter package size
                 // int n_cams = 1;
                 // int inter_package_delay_in_ticks = n_cams * imageSize() * 1.05;
-                cam_->GevSCPD.SetValue(parameters.inter_pkg_delay_);
                 ROS_WARN("Default User Setting Loaded");
             }
         else if (parameters.startup_user_set_ == "UserSet1")
             {
                 cam_->UserSetSelector.SetValue(Basler_GigECameraParams::UserSetSelector_UserSet1);
                 cam_->UserSetLoad.Execute();
-                cam_->GevSCPSPacketSize.SetValue(parameters.mtu_size_);
                 ROS_WARN("User Set 1 Loaded");
             } 
         else if (parameters.startup_user_set_ == "UserSet2")
             {
                 cam_->UserSetSelector.SetValue(Basler_GigECameraParams::UserSetSelector_UserSet2);
                 cam_->UserSetLoad.Execute();
-                cam_->GevSCPSPacketSize.SetValue(parameters.mtu_size_);
                 ROS_WARN("User Set 2 Loaded");
             } 
         else if (parameters.startup_user_set_ == "UserSet3")
             {
                 cam_->UserSetSelector.SetValue(Basler_GigECameraParams::UserSetSelector_UserSet3);
                 cam_->UserSetLoad.Execute();
-                cam_->GevSCPSPacketSize.SetValue(parameters.mtu_size_);
                 ROS_WARN("User Set 3 Loaded");
             } 
         else if (parameters.startup_user_set_ == "CurrentSetting")
             {
-                cam_->GevSCPSPacketSize.SetValue(parameters.mtu_size_);
                 ROS_WARN("No User Set Is selected, Camera current setting will be used");
             }
+
+        ROS_INFO("Setting packet size to %d and packet delay to %d", parameters.mtu_size_, parameters.inter_pkg_delay_);
+        cam_->GevSCPSPacketSize.SetValue(parameters.mtu_size_);
+        cam_->GevSCPD.SetValue(parameters.inter_pkg_delay_);
+
+        if (parameters.ptpEnabled())
+        {
+            // TODO: Code below tries to block until clocks are synced, but it's not airtight.
+            // E.g. consider a case where a master clock is present, and we require 2 cameras
+            // to be slaves. Camera A could register as a master, then camera B could sync to
+            // it as a slave, letting camera B pass this code. Then, camera A syncs to the
+            // desired master clock, leaving the 2 cameras out of sync. This and other edge
+            // cases should resolve within a few seconds, but it's not ideal. One option could
+            // be to have camera nodes communicate between themselves, and all block until all
```

Comment:
Blocking until all conditions are satisfied simultaneously sounds ideal -- how hard would that be to implement?

Reply:
I'm not sure - this is definitely where I have a gap in ROS architecture experience. I'm not sure how each node would know which other nodes or messages to monitor for, and I suspect even then there could still be race conditions.

It's definitely simpler architecture-wise to just pass in the required master clock id, but that's then an extra per-machine config thing to set.

Although, it just occurred to me, if we do a setup where cameras are on separate network interfaces, then it's sufficient to just ensure they are slave clocks and we won't have any issues.

Reply:
We have plenty of other configs -- if you think the best way to do this is to pass in a master clock ID, that's fine on our end (not sure how the library feels about it).

Is there any way you can have one node monitor both cameras, or would that be a total re-write of this software?